src/jllvm/vm/VirtualMachine.cpp

```diff
@@ -182,6 +182,27 @@ jllvm::VirtualMachine::VirtualMachine(std::vector<std::string>&& classPath)
                       // This should have been checked inline in LLVM IR.
                       assert(!classObject->isInitialized());
                       initialize(*classObject);
+                  }},
+        std::pair{"jllvm_build_class_cast_exception", [&](Object* object, ClassObject* classObject)
+                  {
+                      llvm::StringRef className = object->getClass()->getClassName();
+                      llvm::StringRef prefix;
+                      std::string name;
+                      if (classObject->isClass() || classObject->isInterface())
+                      {
+                          prefix = "class";
+                          name = classObject->getClassName();
+                      }
+                      else
+                      {
+                          // TODO: Pretty print array
+                      }
+                      String* string = m_stringInterner.intern(
+                          llvm::formatv("class {0} cannot be cast to {1} {2}", className, prefix, name).str());
```

Comment:
Is `std::format` not yet supported?

Reply:
No in all compilers we care about. GCC's libstdc++ only got an initial implementation in GCC 13 IIRC